src/alr/alr-commands.adb

```diff
@@ -157,6 +160,12 @@ package body Alr.Commands is
                      Long_Switch => "--no-tty",
                      Help        => "Disables control characters in output");
 
+      Define_Switch (Config,
```

Comment:
I like the idea of a Markdown output for the help (see my comment on the implementation in CLIC).

I don't think this should be a global switch though, because it would mean all of the Alire output should be formatted in Markdown which doesn't make sens in my opinion.

Printing help in markdown format is not something users will do by themselves. It's more something that we want to do for the alire.ada.dev website. So for me this should be added to the `alr dev` command (https://github.com/alire-project/alire/blob/master/src/alr/alr-commands-dev.ads).

Something like
```
alr dev --help-doc-makrdown
```
That will print all the help pages for all the commands in markdown format.

Reply:
I agree with you, it doesn't make sense to combine this global switch with any other else command but `help` or the `--help` switches. I thought in using them together when generating the help pages for the website. If you use `--markdown` for anything else, it doesn't have any effect, which is the part where it makes no longer sense.

I wasn't aware of the `dev` subcommand. It makes much sense to add this functionality there, although I have doubts about generating the help pages at once, because that will avoid redirecting each page to a specific markdown file. I already have some code for the website that iterates through all the commands and topics, running `alr --markdown ${topic} --help` and also for `alr --markdown help`. I will look at how could this be done using `alr dev` and reach you again with the changes.